pkg/operator/k8sutil/node.go

```diff
@@ -281,12 +281,18 @@ func GetKubernetesNodesMatchingRookNodes(ctx context.Context, rookNodes []cephv1
 	if err != nil {
 		return nodes, fmt.Errorf("failed to list kubernetes nodes. %+v", err)
 	}
-	for _, kn := range k8sNodes.Items {
-		for _, rn := range rookNodes {
+	for _, rn := range rookNodes {
+		nodeFound := false
+		for _, kn := range k8sNodes.Items {
 			if rookNodeMatchesKubernetesNode(rn, kn) {
 				nodes = append(nodes, kn)
+				nodeFound = true
+				break
 			}
 		}
+		if !nodeFound {
+			logger.Warningf("failed to find matching kubernetes node for %v. Check the CephCluster's config and confirm each 'name' field in spec.storage.nodes matches their 'kubernetes.io/hostname' label", rn)
```

Comment:
When testing, I found this error message. 
```
op-k8sutil: failed to find matching kubernetes node for {foobar {map[] map[] []} map[] {<nil>   [] []}}. 
Check the CephCluster's config and confirm each 'name' field in spec.storage.nodes matches their 'kubernetes.io/hostname' label
```

Let's use the node name.
```suggestion
			logger.Warningf("failed to find matching kubernetes node for %q Check the CephCluster's config and confirm each 'name' field in spec.storage.nodes matches their 'kubernetes.io/hostname' label", rn.Name)
```

Reply:
Thanks for testing it! Fixed the name